diesel_codegen/src/util.rs

```diff
@@ -101,22 +101,29 @@ pub fn inner_of_option_ty(ty: &Ty) -> Option<&Ty> {
     }
 }
 
-pub fn get_options_from_input(attrs: &[Attribute], on_bug: fn() -> !)
-    -> Option<&[MetaItem]>
+pub fn get_options_from_input(attrs: &Vec<Attribute>, on_bug: fn() -> !)
+    -> Option<Vec<MetaItem>>
 {
     let options = attrs.iter().find(|a| a.name() == "options").map(|a| &a.value);
     match options {
-        Some(&MetaItem::List(_, ref options)) => Some(options),
+        Some(&MetaItem::List(_, ref options)) => {
+            Some(options.iter().map(|o| {
+                match o {
+                   &NestedMetaItem::MetaItem(ref m) => m.clone(),
+                   _ => on_bug(),
+                }
+            }).collect())
+        }
```

Comment:
I think this is the change that prompted the `&str` to `String` and `&[..]` to `Vec` changes.

`MetaItem::List` needs to be transformed into a list of `MetaItem`'s. I couldn't figure out how to return a slice of these. Ideas?